system/system_monitor/src/net_monitor/net_monitor.cpp

```diff
@@ -327,6 +338,62 @@ void NetMonitor::check_reassembles_failed(diagnostic_updater::DiagnosticStatusWr
   SystemMonitorUtility::stopMeasurement(t_start, status);
 }
 
+void NetMonitor::check_udp_buf_errors(diagnostic_updater::DiagnosticStatusWrapper & status)
+{
+  // Remember start time to measure elapsed time
+  const auto t_start = SystemMonitorUtility::startMeasurement();
+
+  int whole_level = DiagStatus::OK;
+  std::string error_message;
+
+  uint64_t total_udp_rcvbuf_errors = 0;
+  if (get_value_from_net_snmp(udp_rcvbuf_errors_index_, total_udp_rcvbuf_errors)) {
+    update_delta_queue(
+      udp_rcvbuf_errors_queue_, last_udp_rcvbuf_errors_, total_udp_rcvbuf_errors,
+      udp_buf_errors_check_duration_);
+    uint64_t unit_udp_rcvbuf_errors =
+      std::accumulate(udp_rcvbuf_errors_queue_.begin(), udp_rcvbuf_errors_queue_.end(), 0);
+    status.add(fmt::format("total UDP rcv buf errors"), total_udp_rcvbuf_errors);
+    status.add(fmt::format("UDP rcv buf errors per unit time"), unit_udp_rcvbuf_errors);
+    if (unit_udp_rcvbuf_errors >= udp_buf_errors_check_count_) {
+      whole_level = std::max(whole_level, static_cast<int>(DiagStatus::WARN));
+      error_message += "UDP buf errors";
+    }
+  } else {
+    udp_rcvbuf_errors_queue_.push_back(0);
+    whole_level = std::max(whole_level, static_cast<int>(DiagStatus::ERROR));
+    error_message += "failed to read RcvbufErrors from /proc/net/snmp";
+  }
+
+  uint64_t total_udp_sndbuf_errors = 0;
```

Comment:
It looks like duplicate code, so please try to make it more reusable.

Reply:
@iwatake2222
I'm a bit concerned about the performance impact of reading /proc/net/snmp separately for each of these three items:

- IP fragmentation failed
- UDP send buffer errors
- UDP receive buffer errors

I think it would be better to consolidate these `std::ifstream ifs("/proc/net/snmp");` into a single function. (it is just an idea that creating a class including NetSnmp instances. the design is up to you.).

Reply:
@TetsuKawa 
Thank you for your feedback.  
I had exactly the same thought during the implementation.  

However, I concluded that there is no performance concern for the following reasons:
- generally, the overhead of accessing `/proc` is smaller than accessing a normal file
- the actual measurement results showed no noticeable increase in processing load

Of course, there might be a very very slight increase that is not visible at this level, but I judged that the current implementation is the better approach considering the following factors:
- Increased complexity of the implementation
- Higher maintenance costs if the number of monitored metrics increases in the future
- Increased risk of regression (with potential impacts on all `/proc/net/snmp`-related information)  

Therefore, I would like to keep the current design and implementation as it is :pray:

Reply:
I understand. 
As you say, since /proc exists in memory rather than on SSD, it probably doesn't increase the process load much. (This is very educational for me.) 
I agree with your thinking!